File: app/src/androidTest/java/ch/epfl/sdp/healthplay/scan/ProductManagerTest.java
package ch.epfl.sdp.healthplay.scan;

import static org.junit.Assert.*;

import androidx.annotation.VisibleForTesting;

import org.junit.Test;

import java.io.IOException;

import ch.epfl.sdp.healthplay.model.ProductInfoClient;

public class ProductManagerTest {

    @Test
    public void getProductManager() {
        assertFalse(ProductManager.getProduct("asdfasdf").join().isPresent());
    }

    @Test
    public void getCorrectProductManager() {
        assertTrue(ProductManager.getProduct("737628064502").join().isPresent());
    }

    @Test
    public void withExceptionProductManager() {
        assertFalse(ProductManager.getProductInner(() -> {
            throw new Exception();
        }).join().isPresent());
    }
}

File: app/src/androidTest/java/ch/epfl/sdp/healthplay/scan/ProductManagerTest.kt
package ch.epfl.sdp.healthplay.scan

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import java.lang.Exception

class ProductManagerTest {
    @Test
    fun productManager() {
        assertFalse(ProductManager.getProduct("asides").join().isPresent)
    }

    @Test
    fun correctProductManager() {
        assertTrue(ProductManager.getProduct("737628064502").join().isPresent)
    }

    @Test
    fun withExceptionProductManager() {
        assertFalse(ProductManager.getProductInner { throw Exception() }.join().isPresent)
    }
}